### PR TITLE
openimageio 2.3.18.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by cho-m** [comment]
I think `jpeg` can be migrated to `jpeg-turbo` here.

Can't migrate to `python@3.10` due to dependency tree. There is separate PR to handle this in #106949.

**Comment by cho-m** [comment]
Seems like it is having trouble with `ffmpeg`. Support for v5.0 should be there since 2.3.13 https://github.com/OpenImageIO/oiio/releases/tag/v2.3.13.0.

```
CMake Error at src/cmake/modules/FindFFmpeg.cmake:64 (string):
    string sub-command REGEX, mode MATCHALL needs at least 5 arguments total to
    command.
```

**Comment by cho-m** [comment]
https://github.com/OpenImageIO/oiio/pull/3507

File: Formula/openimageio.rb [hunk 1]
```diff
@@ -47,7 +47,7 @@ def install
       -DCCACHE_FOUND=
       -DEMBEDPLUGINS=ON
       -DUSE_FIELD3D=OFF
```
Comment (cho-m) [inline Formula/openimageio.rb:49]: Unrelated to current failure, but we may want to build with Homebrew `pugixml` instead:
```suggestion
      -DUSE_EXTERNAL_PUGIXML=ON
      -DUSE_FIELD3D=OFF
```

Ref: https://github.com/OpenImageIO/oiio/blob/v2.3.18.0/INSTALL.md#optional-dependencies----features-may-be-disabled-if-not-found
> - We use PugiXML for XML parsing. There is a version embedded in the OIIO tree, but if you want to use an external, system-installed version (as may be required by some software distributions with policies against embedding other projects), then just build with `-DUSE_EXTERNAL_PUGIXML=1`. Any PugiXML >= 1.8 should be fine (we have tested through 1.11).

**Comment by carlocab** [comment]
Can we roll this into https://github.com/Homebrew/homebrew-core/pull/106949?

**Comment by carlocab** [comment]
Needs rebase for https://github.com/Homebrew/homebrew-core/pull/106949.

**Comment by cho-m** [comment]
Fastest way to merge is to go back to `ffmpeg@4` as FFmpeg 5.0 is supported but build has trouble detecting FFmpeg 5.1.

Something like following may work for `ffmpeg >= 5.1` on top of https://github.com/FFmpeg/FFmpeg/commit/f2da2e1458b76a1d6c068673430b46cf2850bc51:
```diff
diff --git a/src/cmake/modules/FindFFmpeg.cmake b/src/cmake/modules/FindFFmpeg.cmake
index bb1e9d65..0e5ac297 100644
--- a/src/cmake/modules/FindFFmpeg.cmake
+++ b/src/cmake/modules/FindFFmpeg.cmake
@@ -59,7 +59,11 @@ else ()
 endif ()

 if (FFMPEG_INCLUDES)
-  file(STRINGS "${FFMPEG_INCLUDES}/libavcodec/version.h" TMP
+  set (_libavcodec_version_major_h "${FFMPEG_INCLUDES}/libavcodec/version_major.h")
+  if (NOT EXISTS "${_libavcodec_version_major_h}")
+    set (_libavcodec_version_major_h "${FFMPEG_INCLUDES}/libavcodec/version.h")
+  endif()
+  file(STRINGS "${_libavcodec_version_major_h}" TMP
        REGEX "^#define LIBAVCODEC_VERSION_MAJOR .*$")
   string (REGEX MATCHALL "[0-9]+[.0-9]+" LIBAVCODEC_VERSION_MAJOR "${TMP}")
   file(STRINGS "${FFMPEG_INCLUDES}/libavcodec/version.h" TMP
@@ -69,7 +73,9 @@ if (FFMPEG_INCLUDES)
        REGEX "^#define LIBAVCODEC_VERSION_MICRO .*$")
   string (REGEX MATCHALL "[0-9]+[.0-9]+" LIBAVCODEC_VERSION_MICRO "${TMP}")
   set (LIBAVCODEC_VERSION "${LIBAVCODEC_VERSION_MAJOR}.${LIBAVCODEC_VERSION_MINOR}.${LIBAVCODEC_VERSION_MICRO}")
-  if (LIBAVCODEC_VERSION VERSION_GREATER_EQUAL 59.18.100)
+  if (LIBAVCODEC_VERSION VERSION_GREATER_EQUAL 59.37.100)
+      set (FFMPEG_VERSION 5.1)
+  elseif (LIBAVCODEC_VERSION VERSION_GREATER_EQUAL 59.18.100)
       set (FFMPEG_VERSION 5.0)
   elseif (LIBAVCODEC_VERSION VERSION_GREATER_EQUAL 58.134.100)
       set (FFMPEG_VERSION 4.4)
```

File: Formula/openimageio.rb [hunk 2]
```diff
@@ -62,22 +81,6 @@ def install
...
-      python.opt_frameworks/"Python.framework/Versions"/py3ver
-    else
-      python.opt_prefix
-    end
-
-    ENV["PYTHONPATH"] = prefix/Language::Python.site_packages(python3)
-
-    args << "-DPython_EXECUTABLE=#{python3}"
-    args << "-DPYTHON_LIBRARY=#{py3prefix}/lib/#{shared_library("libpython#{py3ver}")}"
-    args << "-DPYTHON_INCLUDE_DIR=#{py3prefix}/include/python#{py3ver}"
```
Comment (cho-m) [inline Formula/openimageio.rb:78]: No longer used probably after changes like https://github.com/OpenImageIO/oiio/pull/2705 and https://github.com/OpenImageIO/oiio/pull/2764
```
  Manually-specified variables were not used by the project:

    PYTHON_INCLUDE_DIR
    PYTHON_LIBRARY
    USE_FIELD3D
```

Also, Field3d support was deprecated and isn't used unless `-DENABLE_FIELD3D`

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**